lib/components/UploadPicker.vue

```diff
@@ -240,54 +300,116 @@ export default Vue.extend({
+		async getContent(path?: string) {
+			return Array.isArray(this.content) ? this.content : await this.content(path)
+		},
+
+		/**
+		 * Show a dialog to let the user decide how to proceed with invalid filenames.
+		 * The returned promise resolves to true if the file should be renamed and resolves to false to skip it the file.
+		 * The promise rejects when the user want to abort the operation.
+		 *
+		 * @param filename The invalid file name
+		 */
+		async showInvalidFileNameDialog(filename: string) {
+			return new PCancelable(async (resolve, reject) => {
+				await new DialogBuilder()
+					.setName(t('Invalid file name'))
+					.setSeverity('error')
+					.setText(t('"{filename}" contains invalid characters, how do you want to continue?', { filename }))
+					.setButtons([
+						{
+							label: t('Rename'),
+							type: 'primary',
+							callback: () => resolve(true),
+						},
+						{
+							label: t('Skip'),
+							callback: () => resolve(false),
+						},
+						{
+							label: t('Cancel'),
+							type: 'error',
+							callback: reject,
+						},
+					])
+					.build()
+					.show()
+			})
+		},
+
+		async handleConflicts(nodes: Array<File|Directory>, path: string): Promise<Array<File|Directory>|false> {
+			const invalidReplacement = ['-', '_', ' '].filter((c) => !this.forbiddenCharacters.includes(c))[0] ?? 'x'
+
+			try {
+				const content = path === '' ? this.currentContent : await this.getContent(path).catch(() => [])
+				const conflicts = getConflicts(nodes, content)
+
+				// First handle conflicts as this might already remove invalid files
+				if (conflicts.length > 0) {
+					const { selected, renamed } = await openConflictPicker(path, conflicts, content, { recursive: true })
+					nodes = [...selected, ...renamed]
 				}
-			}
 
-			files.forEach(file => {
-				const forbiddenCharacters = (this.forbiddenCharacters || []) as string[]
-				const forbiddenChar = forbiddenCharacters.find(char => file.name.includes(char))
-
-				if (forbiddenChar) {
-					showError(t(`"${forbiddenChar}" is not allowed inside a file name.`))
-				} else {
-					this.uploadManager.upload(file.name, file)
-						.catch(() => {
-							// Ignore errors, they are handled by the upload manager
-						})
+				// We need to check all files for invalid characters
+				const filesToUpload: Array<File|Directory> = []
+				for (const file of nodes) {
+					const invalid = this.forbiddenCharacters.some((c) => file.name.includes(c))
+					// No invalid characters used on this file, so just continue
+					if (!invalid) {
+						filesToUpload.push(file)
+						continue
+					}
+
+					// Hanle invalid path
+					if (await this.showInvalidFileNameDialog(file.name)) {
+						// create a new valid path name
+						let newName = file.name
+						this.forbiddenCharacters.forEach((c) => { newName = newName.replaceAll(c, invalidReplacement) })
+						Object.defineProperty(file, 'name', { value: newName })
+						filesToUpload.push(file)
+					}
```

Comment:
Maybe more readable:

```suggestion
					if (invalid) {
						// Hanle invalid path
						if (await this.showInvalidFileNameDialog(file.name)) {
							// create a new valid path name
							let newName = file.name
							this.forbiddenCharacters.forEach((c) => { newName = newName.replaceAll(c, invalidReplacement) })
							Object.defineProperty(file, 'name', { value: newName })
							filesToUpload.push(file)
						} else {
							continue
						}
					}

					filesToUpload.push(file)

```

Reply:
My idea was to have only one `filesToUpload.push(file)`, but honestly feel free to ignore as this is superfluous ;)